prjxray/timing.py

```diff
@@ -0,0 +1,104 @@
+""" Route timing delay definitions.
+
+Routing delay is formed from two parts in this model:
+
+- Intristic delay of the element
+- Capactive loading delay of the net
+
+Intristic delay is a time value (e.g. nanoseconds), does not vary based on
+routing fanout.  It does vary based on the PVT (process, voltage, temperature)
+corner.  PvtCorner and IntristicDelay objects are used to model intristic
+delay of elements.
+
+Capactive loading is the elmore delay from the RC tree formed by interconnect.
+The RC tree is made up of 5 types of RC nodes:
+
+1. Out-pins
+2. Buffer switches
+3. Pass-through switches
+4. Wires
+5. In-pins
+
+Outpins have intristic delays and an driver resistence.
+
+Buffer switches have instrinsic delays, internal capacitance, and output driver
+resistence.
+
+Pass-through switches have instrinsic delays and a resistence and capacitance.
+
+Wires have a resistence and capacitance.
+
+Inpins have instrinsic delays and a sink capacitance.
+
+The elmore delay is the RC tree formed by these 5 components.  Out pins and
+buffer switches are the roots of the elmore tree. Buffer switches
+and inpins are leafs of the elmore tree. Wires and pass-through switches are
+nodes in the tree, and share their capacitance upstream and downstream using a
+pi-model.
+
+"""
```

Comment:
It might be worth using [`type hints`](https://docs.python.org/3/library/typing.html)? See example
https://github.com/SymbiFlow/symbiflow-arch-defs/blob/0c856ca4c89645b7d1866cdf77ce36a9366eddde/utils/lib/pb_type.py#L4-L32